src/io/simulator/simulator_handle.hpp

```diff
@@ -122,12 +122,13 @@ class SimulatorHandle {
     while (!should_shut_down_ && (cluster_wide_time_microseconds_ < deadline)) {
       if (can_receive_.contains(receiver)) {
         std::vector<OpaqueMessage> &can_rx = can_receive_.at(receiver);
-        if (!can_rx.empty()) {
-          OpaqueMessage message = std::move(can_rx.back());
-          can_rx.pop_back();
-
-          // TODO(tyler) search for item in can_receive_ that matches the desired types, rather
-          // than asserting that the last item in can_rx matches.
```

Comment:
We probably want to keep the old code
If we get nullopt, splog error or mg assert: "You are trying to receive message while not giving correct type"